### PR TITLE
openthread: Enable nrf_security support for precompiled OT libs

### PR DESCRIPTION
**Pull request by lmaciejonczyk**
From this commit nrf_security can be used together with precompiled OpenThread libraries.

- [x] it is blocked by: https://github.com/nrfconnect/sdk-nrfxlib/pull/268
- [x] needs updates after https://github.com/nrfconnect/sdk-nrfxlib/pull/242 is in

### CODE REVIEW

**Review by edmont — APPROVED** [review]

File: doc/nrf/ug_thread_configuring.rst [hunk 1]
```diff
@@ -206,6 +206,9 @@ You can use these sets for configuration purposes or for building libraries with
 
 Selecting these sets is not related to :ref:`thread_ug_device_type`.
 
+.. note::
+    Notice that while building OpenThread from sources an additional user configuration can overwrite selected sets of features.
```
Comment (pszkotak) [inline doc/nrf/ug_thread_configuring.rst:210]: Can we be more strict here?
I propose something like that:
```suggestion
    While building OpenThread from sources the user configuration overwrites selected predefined Master, FTD or MTD feature set.
```

File: doc/nrf/ug_thread_configuring.rst [hunk 2]
```diff
@@ -312,6 +315,9 @@ After the sample is build execute the following to overwrite the exising nrfxlib
 
   west build -t install_openthread_libraries
 
+.. note::
+    The libraries destination directory differ based on chosen option :option:`CONFIG_OPENTHREAD_NORDIC_LIBRARY` and cryptography backends: :option:`CONFIG_CC310_BACKEND` or :option:`CONFIG_OBERON_BACKEND`.
```
Comment (pszkotak) [inline doc/nrf/ug_thread_configuring.rst:319]: Awesome!

**Review by pszkotak — APPROVED** [review]
LGTM, thanks

File: doc/nrf/ug_thread_configuring.rst [hunk 3]
```diff
@@ -58,7 +68,7 @@ This includes the following options:
   You can set any value ranging from ``0`` to ``65535``.
 
 Default configuration reference
-    The default configuration for all :ref:`openthread_samples` is defined in :file:`nrf/samples/openthread/openthread.conf`.
+    The default configuration for all :ref:`openthread_samples` is defined in :file:`nrf/samples/common/overlay-ot-defaults.conf`.
```
Comment (lmaciejonczyk) [inline doc/nrf/ug_thread_configuring.rst:71]:
```suggestion
    The default configuration for all :ref:`openthread_samples` is defined in :file:`nrf/samples/openthread/common/overlay-ot-defaults.conf`.
```

File: doc/nrf/ug_thread_configuring.rst [hunk 4]
```diff
@@ -307,16 +320,20 @@ The following table lists the supported features for each of these sets.
       - ✔
       -
 
-Replacing OpenThread Libraries
-******************************
+Replacing OpenThread libraries
+==============================
+
+Each OpenThread sample provides a possibility to replace nrfxlib's :ref:`nrfxlib:ot_libs` when configured to build OpenThread stack from source with :option:`CONFIG_OPENTHREAD_SOURCES`.
+After the sample is built, run the following command in the sample directory to overwrite the existing nrfxlib libraries with the new ones:
```
Comment (lmaciejonczyk) [inline doc/nrf/ug_thread_configuring.rst:327]: Actually you don't need to run the command always in the sample directory. For instance you can just run:
`west build -b nrf52840dk_nrf52840 -p always nrf/samples/openthread/cli/` from root directory and there you can just run `west build -t install_openthread_libraries` without moving to sample directory.

File: doc/nrf/ug_thread_configuring.rst [hunk 5]
```diff
@@ -198,7 +208,10 @@ OpenThread configuration sets
 *****************************
 
 |NCS| provides predefined sets of features and optional functionalities from the OpenThread stack.
-You can use these sets for configuration purposes or for building libraries with complete Thread specification support.
+These predefined sets are based on :ref:`nrfxlib:ot_libs` available in nrfxlib.
```
Comment (lmaciejonczyk) [inline doc/nrf/ug_thread_configuring.rst:211]: I would rather say that :ref:`nrfxlib:ot_libs` provide functionalities listed in these predefined sets.

File: doc/nrf/ug_thread_configuring.rst [hunk 6]
```diff
@@ -198,7 +208,10 @@ OpenThread configuration sets
...
+You can use these sets for configuration purposes or for building libraries with complete Thread specification support when you :ref:`configure OpenThread to use pre-built libraries <ug_thread_configuring_basic_building>` with the :option:`CONFIG_OPENTHREAD_LIBRARY_1_1` Kconfig option.
```
Comment (lmaciejonczyk) [inline doc/nrf/ug_thread_configuring.rst:212]:
```suggestion
You can use these sets for configuration purposes when you build your application using OpenThread sources or for building application with complete Thread specification support when you :ref:`configure OpenThread to use pre-built libraries <ug_thread_configuring_basic_building>` with the :option:`CONFIG_OPENTHREAD_LIBRARY_1_1` Kconfig option.
```

**Comment by greg-fer** [comment]
I discussed documentation with @lmaciejonczyk  and we agreed we're going to come back to the topic after he makes some changes to the menuconfig structure.

File: doc/nrf/ug_thread_configuring.rst [hunk 7]
```diff
@@ -304,13 +322,24 @@ The following table lists the supported features for each of these sets.
...
 
-  west build -t install_openthread_libraries
+.. parsed-literal::
+   :class: highlight
+
+   west build -t install_openthread_libraries
+
+.. note::
+    The libraries destination directory can differ.
+    When you selected :option:`CONFIG_OPENTHREAD_USER_CUSTOM_LIBRARY`, the location depends the chosen nRF Security backend, either :option:`CONFIG_CC310_BACKEND` or :option:`CONFIG_OBERON_BACKEND`.
```
Comment (MarekPorwisz) [inline doc/nrf/ug_thread_configuring.rst:342]: Something to consider: Maybe we should not mention backends here and mention it "depends on device hardware crypto capabilities" as it is more complex than just CC and OBERON i.e. on nRF53 we'll have CC312, and on 52833 currently we use mixed config oberon + vanilla due to lack of SHA224 support in oberon(to be added soon).

**Review by MarekPorwisz — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]

File: samples/openthread/cli/prj.conf [hunk 8]
```diff
@@ -14,5 +14,4 @@ CONFIG_SHELL_ARGC_MAX=26
 CONFIG_SHELL_CMD_BUFF_SIZE=416
 
 # Enable OpenThread features set
-CONFIG_OPENTHREAD_NORDIC_LIBRARY=y
```
Comment (edmont) [inline samples/openthread/cli/prj.conf:17]: Same thing applies to the NCP sample.

**Review by tejlmand — APPROVED** [review]

File: doc/nrf/ug_thread_configuring.rst [hunk 9]
```diff
@@ -225,98 +252,155 @@ The following table lists the supported features for each of these sets.
...
+This option is disabled by default.
+
+Run the following command to update the nrfxlib libraries:
+
+.. parsed-literal::
+   :class: highlight
+
+   west build -b nrf52840dk_nrf52840 -t install_openthread_libraries -- -DOPENTHREAD_BUILD_OUTPUT_STRIPPED=y
+   This command builds two versions of the libraries, with and without debug symbols, and installs only the version without debug symbols.
+
```
Comment (greg-fer) [inline doc/nrf/ug_thread_configuring.rst:388]:
```suggestion

This command builds two versions of the libraries, with and without debug symbols, and installs only the version without debug symbols.
```

**Review by greg-fer — CHANGES_REQUESTED** [review]

**Comment by tejlmand** [comment]
please rebase

**Review by greg-fer — APPROVED** [review]

**Comment by rlubos** [comment]
Doc build failed with quite a few complaints:
https://jenkins-ncs.nordicsemi.no/job/latest/job/sub/job/test-fw-nrfconnect-nrf_doc/job/master/6029/artifact/_build/nrfxlib/doc.log

**Comment by lmaciejonczyk** [comment]
How can we disable CI check for `nrf_security_mbedtls_configuration.h` files which exist only for checking OT libs dependencies?

**Comment by rlubos** [comment]
@utzig @ru-fu Can particular header files be excluded from documentation build? Some mbedTLS config files are generating warnings, thus causing CI failure:
https://jenkins-ncs.nordicsemi.no/job/latest/job/sub/job/test-fw-nrfconnect-nrf_doc/job/master/6050/artifact/_build/nrfxlib/doc.log

**Comment by lmaciejonczyk** [comment]
I can change the file extension if it helps.

**Comment by ru-fu** [comment]
> 
> 
> @utzig @ru-fu Can particular header files be excluded from documentation build? Some mbedTLS config files are generating warnings, thus causing CI failure:
> https://jenkins-ncs.nordicsemi.no/job/latest/job/sub/job/test-fw-nrfconnect-nrf_doc/job/master/6050/artifact/_build/nrfxlib/doc.log

Yes, you can exclude files or folders in the nrf/doc/nrfxlib/nrfxlib.doxyfile.in file (line 805).

**Comment by utzig** [comment]
> > @utzig @ru-fu Can particular header files be excluded from documentation build? Some mbedTLS config files are generating warnings, thus causing CI failure:
> > https://jenkins-ncs.nordicsemi.no/job/latest/job/sub/job/test-fw-nrfconnect-nrf_doc/job/master/6050/artifact/_build/nrfxlib/doc.log
> 
> Yes, you can exclude files or folders in the nrf/doc/nrfxlib/nrfxlib.doxyfile.in file (line 805).

If this include file has a proper include guard you can also try pre-defining it for Doxygen here: https://github.com/nrfconnect/sdk-nrf/blob/master/doc/nrfxlib/nrfxlib.doxyfile.in#L1947

**Review by MarekPorwisz — APPROVED** [review]
Tested and seems to work

File: doc/nrf/ug_thread_configuring.rst [hunk 10]
```diff
@@ -233,98 +260,155 @@ The following table lists the supported features for each of these sets.
...
+=======================================
 
-Replacing OpenThread Libraries
-******************************
+You can update nrfxlib's :ref:`nrfxlib:ot_libs` when using any Thread sample if you configure the sample to build OpenThread stack from source with :option:`CONFIG_OPENTHREAD_SOURCES`.
+Use this functionality for example for :ref:`certification <ug_thread_cert>` of your configuration of OpenThread libraries.
+
+.. note::
+    The libraries destination directory can differ.
+    When you selected :option:`CONFIG_OPENTHREAD_USER_CUSTOM_LIBRARY`, the location depends the chosen :ref:`nrf_security backend <nrfxlib:nrf_security_readme>`, either :option:`CONFIG_CC310_BACKEND` or :option:`CONFIG_OBERON_BACKEND`.
```
Comment (frkv) [inline doc/nrf/ug_thread_configuring.rst]: CONFIG_CC3XX_BACKEND

**Review by frkv — APPROVED** [review]

**Pull request merged**